todo-frontend-angular/src/app/todolist-selector/add-todo-modal/add-todo-modal.component.ts

```diff
@@ -0,0 +1,53 @@
+import { Component, EventEmitter, Output, Input } from '@angular/core';
+import { FormControl,Validators, FormGroup } from '@angular/forms';
+import { NewTodoDTO } from 'src/app/models/new-todo-DTO.model';
+
+
+@Component({
+  selector: 'app-add-todo-modal',
+  templateUrl: './add-todo-modal.component.html',
+  styleUrls: ['./add-todo-modal.component.scss']
+})
+export class AddTodoModalComponent {
+  isVisible = false;
+  isOkLoading = false;
+  validateForm: FormGroup<{
+    title: FormControl<string | null>;
+    datePicker: FormControl<Date | null>;
+}> = new FormGroup({
+  title: new FormControl<string | null>(null, [Validators.required, Validators.minLength(1)]),
+  datePicker: new FormControl<Date | null>(null,[Validators.required,Validators.minLength(8)])
+}) 
+constructor(){
+
+}
+  showModal(): void {
+    this.isVisible = true;
+  }
+  @Input() listId?: number
+  
+  @Output()
+  Submit=new EventEmitter<NewTodoDTO>();
```

Comment:
remember that Submit can sometimes be a reserved keyword :D maybe a better naming would be addTodoClick? or addTodoSubmit or todoSubmit?